src/SimpleAjax/SimpleAjax.php

```diff
@@ -82,31 +81,41 @@ public function __construct()
         \Controller::setStaticUrls();
     }
 
-
     /**
      * Handle the request
      */
     public function handle()
     {
         global $container;
+
         /** @var EventDispatcher $dispatcher */
         $dispatcher = $container['event-dispatcher'];
 
         // Trigger event
+        // @since 1.1
         $event = new SimpleAjaxEvent($this->isIncludeFrontendExclusive());
         $dispatcher->dispatch(SimpleAjaxEvent::NAME, $event);
 
+        // If the event listener does not terminate the process by itself, check for a response instance to send
+        // @since 1.2
+        $response = $event->getResponse();
+        if (null !== $response) {
+            $response->send();
+        }
+
         // Run hooks
         $this->runHooks();
 
         // If there is no other output, we generate a 412 error response
-        header('HTTP/1.1 412 Precondition Failed');
-        die('Simple Ajax: Invalid AJAX call.');
+        $errorResponse = new Response(
+            'Simple Ajax: Invalid AJAX call.',
+            Response::HTTP_PRECONDITION_FAILED
+        );
+        $errorResponse->send();
     }
 
-
     /**
-     * Run hooks (to be removed)
+     * Run hooks (legacy)
      */
     private function runHooks()
     {
```

Comment:
I wonder if you want to deprecate the HOOKs and trigger an `E_USER_DEPRECATED` here if the array is not empty?